### PR TITLE
feat: add long word validation for name input in onboarding and settings

### LINKED ISSUE #1280: UI: Prevent navbar layout breaking from excessively long user names
## Description

Currently, the user's display name appears in the navbar after onboarding and can also be updated later from the settings page.

There is no effective handling for excessively long names, which can cause the navbar layout to break. Since the name is positioned on the top-right section of the navbar, a very long name pushes other navbar elements (such as the search bar and application title) out of alignment and toward the left side of the screen.

In extreme cases, a user can enter a very long continuous string, causing the name field to occupy a large portion of the navbar width.

## Expected Behavior

The navbar should remain stable and responsive regardless of the user's display name length.

Suggested improvements:

- Add a reasonable maximum character limit for the user name field during onboarding and settings update.
- Prevent overly long continuous strings from breaking the layout.
- Truncate long names in the navbar using ellipsis (`...`) once the available width is exceeded.

Example:

- `Jonathan Christopher Anderson`
  → `Jonathan...`
  or better yet,
  → `Jonathan` only.

Whichever looks very UI friendly

## Possible Implementation Ideas

- Use CSS text truncation techniques such as:
  - `overflow: hidden`
  - `text-overflow: ellipsis`
  - `white-space: nowrap`
- Add frontend validation for maximum allowed name length.
- Ensure the navbar remains responsive across different screen sizes.

## Acceptance Criteria

- Long names no longer shift or break navbar elements.
- Navbar layout remains responsive and visually consistent.
- Truncated names display properly with ellipsis.
- User input validation prevents excessively long names.
- Works correctly on both desktop and smaller screen widths.

## Additional Notes

Please ensure the solution is implemented consistently for:
- Initial onboarding flow
- Settings/profile name update flow
- All navbar variants where the user name is displayed

### PR DESCRIPTION
**Pull request by Umang-Khemka**
## Summary
The following changes have been made to fix the navbar layout breaking due to long user names:

- Truncate long names in the navbar showing only the first name with full name visible on hover
- Added word-length validation (max 30 chars per word) in onboarding to prevent continuous strings
- Added word-length validation (max 30 chars per word) in settings to prevent continuous strings
- Show error message when a continuous string exceeds 30 characters

## Addressed Issues:
Fixes #1280

## Screenshots:
Before:

Screenshot 2026-05-24 201234

After:

Screenshot 2026-05-24 203000

## Additional Notes:
Changes are limited to 3 frontend files only:
- `frontend/src/components/Navbar.tsx`
- `frontend/src/components/OnboardingSteps/AvatarSelectionStep.tsx`
- `frontend/src/pages/SettingsPage/components/AccountSettingsCard.tsx`

- [x] This PR does not contain AI-generated code at all.
- [ ] This PR contains AI-generated code. I have read the [AI Usage Policy](https://github.com/AOSSIE-Org/Info/blob/main/AI-UsagePolicy.md) and this PR complies with this policy. I have tested the code locally and I am responsible for it.

## Checklist
- [x] My PR addresses a single issue, fixes a single bug or makes a single improvement.
- [x] My code follows the project's code style and conventions
- [x] If applicable, I have made corresponding changes or additions to the documentation
- [ ] If applicable, I have made corresponding changes or additions to tests
- [x] My changes generate no new warnings or errors
- [x] I have joined the [Discord server](https://discord.gg/hjUhu33uAn) and I will share a link to this PR with the project maintainers there
- [x] I have read the [Contribution Guidelines](../CONTRIBUTING.md)
- [x] Once I submit my PR, CodeRabbit AI will automatically review it and I will address CodeRabbit's comments.
- [x] I have filled this PR template completely and carefully, and I understand that my PR may be closed without review otherwise.

## Summary by CodeRabbit

* **New Features**
  * Added name field validation that prevents saving names containing words longer than 30 characters, with inline error messaging in profile setup and account settings.

* **Improvements**
  * Updated welcome message display to show only the user's first name, with the full name available as a tooltip on hover.

[Review Change Stack](https://app.coderabbit.ai/change-stack/AOSSIE-Org/PictoPy/pull/1286?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: frontend/src/pages/SettingsPage/components/AccountSettingsCard.tsx [hunk 1]
```diff
@@ -72,12 +78,16 @@ const AccountSettingsCard: React.FC = () => {
...
+              className={`h-10 w-full text-sm placeholder:text-sm ${nameError
+                ? 'border-red-500 placeholder:text-red-500/80 focus-visible:ring-red-500'
+                : ''
+                }`}
             />
+            {longWordError && (
+              <p className="mt-2 text-xs text-red-500">
+                A single word in your name cannot exceed 30 characters.
+              </p>
+            )}
```
[the remark this answers is not in the document]
Comment (Umang-Khemka) [reply frontend/src/pages/SettingsPage/components/AccountSettingsCard.tsx:91]: Fixed. The Save button is now disabled when longWordError is active.

File: frontend/src/components/Navigation/Navbar/Navbar.tsx [hunk 2]
```diff
@@ -81,7 +81,13 @@ export function Navbar() {
...
         <div className="flex items-center space-x-2">
           <span className="hidden text-sm sm:inline-block">
-            Welcome <span className="text-muted-foreground">{userName}</span>
+            Welcome{' '}
+            <span
+              className="text-muted-foreground inline-block max-w-[150px] truncate align-bottom"
+              title={userName}
+            >
+              {userName.split(' ')[0]}
+            </span>
```
[the remark this answers is not in the document]
Comment (Umang-Khemka) [reply frontend/src/components/Navigation/Navbar/Navbar.tsx:90]: Fixed. Added optional chaining to guard userName before calling split() — changed to userName?.split(' ')[0] ?? '' to prevent runtime crashes when userName is null or undefined.

File: frontend/src/components/OnboardingSteps/AvatarSelectionStep.tsx [hunk 3]
```diff
@@ -56,6 +57,13 @@ export const AvatarSelectionStep: React.FC<AvatarNameSelectionStepProps> = ({
...
 
   const handleNameChange = (value: string) => {
+    const words = value.split(' ');
+    const hasLongWord = words.some((word) => word.length > 30);
+    if (hasLongWord) {
+      setLongWordError(true);
+      return;
+    }
+    setLongWordError(false);
     setLocalName(value);
```
[the remark this answers is not in the document]
Comment (Umang-Khemka) [reply frontend/src/components/OnboardingSteps/AvatarSelectionStep.tsx:67]: Acknowledged. Adding automated tests for the word-length validation logic (boundary cases 30/31 chars, multi-space input, recovery after invalid input) will be addressed in a follow-up PR to keep this PR focused on the core navbar layout fix.

[the remark this answers is not in the document]
Comment (rohan-pandeyy) [reply frontend/src/components/OnboardingSteps/AvatarSelectionStep.tsx:67]: @coderabbitai yes. open the issue

**Review by rohan-pandeyy — APPROVED** [review]

**Comment by rohan-pandeyy** [comment]
Thank you for your contribution @Umang-Khemka !
